src/Pos/Wallet/Web/ClientTypes.hs

```diff
@@ -101,18 +112,25 @@ instance Hashable CHash where
     hashWithSalt s (CHash h) = hashWithSalt s h
 
 -- | Client address
-newtype CAddress = CAddress CHash deriving (Show, Eq, Generic, Hashable, Buildable)
+-- @w@ is phantom type and stands for type of item this address belongs to.
+newtype CAddress w = CAddress CHash deriving (Show, Eq, Generic, Hashable, Buildable)
+
+-- | Marks address as belonging to wallet set.
+data WS = WS
+
+-- | Marks address as belonging to account.
+data Acc = Acc
```

Comment:
This distinction doesn't allow you to put address of walletset to here, for instance https://github.com/input-output-hk/cardano-sl/pull/145/files/7e1c57ed2f83f67399e03ad330549378363df829#diff-1bdbe78996ec3f2f9dfc5d81aa33b903R179